test/object-store/realm.cpp

```diff
@@ -1206,6 +1207,102 @@ TEST_CASE("Get Realm using Async Open", "[sync][pbs][async open]") {
         REQUIRE(got_error);
     }
 
+    SECTION("waiters are cancelled if cancel_waits_on_nonfatal_error") {
+        auto logger = util::Logger::get_default_logger();
+        auto transport = std::make_shared<HookedUnitTestTransport>();
+        auto socket_provider = std::make_shared<HookedSocketProvider>(logger, "some user agent");
+        enum TestMode { location_fails, token_fails, token_not_authorized };
+
+        OfflineAppSession::Config oas_config;
+        oas_config.transport = transport;
+        oas_config.socket_provider = socket_provider;
+        OfflineAppSession oas(oas_config);
+
+        SyncTestFile config(oas, "realm");
+        auto user = config.sync_config->user;
+        config.sync_config->cancel_waits_on_nonfatal_error = true;
```

Comment:
should we add a test (maybe already have one?) when this is false?

Reply:
yeah, what I meant is to test that behavior and expectations too. But I think you added a test in the previous PR.

Reply:
Yes, the tests added in the previous PR should test the normal cases; although I could add an additional test specifically for async open if you think that is necessary.